### PR TITLE
Update spec for file support

### PR DESCRIPTION
**Pull request by mjcuva**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -32,15 +32,40 @@ else:
 **Node:**
 
 ```node
-var api = require('api').config('API_KEY');
+let api = require('api').config('API_KEY');
```
Comment (domharrington) [inline README.md]: Can we make this a const?

File: README.md [hunk 2]
```diff
@@ -32,15 +32,40 @@ else:
...
 
 api('math').run('multiply', {
     'x': 1,
```
Comment (domharrington) [inline README.md]: Real nit, but can we make this 2 spaces?

File: README.md [hunk 3]
```diff
@@ -32,15 +32,40 @@ else:
...
     'y': 2,
 }, function(err, res) {
   if(err) return console.log(err);
```
Comment (domharrington) [inline README.md]: Can you add a space between `if` and `(err)`

File: README.md [hunk 4]
```diff
@@ -32,15 +32,40 @@ else:
...
+Build services support invoking with files as data. Invoking with a stream, buffer, or parameter to `api.file` (detailed below) should be supported. Responses containing buffers might also need to be parsed to the appropriate data type for the specific language.
+
+### api.file
+
+`api.file` should be able to take a local path to a file or a url, and convert it to a format that can be sent to build (stream or buffer).
+
+Example (Node):
+
+```node
+let api = require('api').config('API_KEY');
```
Comment (domharrington) [inline README.md]: Can you const this too?

File: README.md [hunk 5]
```diff
@@ -32,15 +32,40 @@ else:
...
+
+api('file-test').run('scale', { image: api.file('https://files.readme.io/88d46f2-small-logo.png'), width: 1000 }, (err, response) => {
```
Comment (domharrington) [inline README.md]: Does this API exist?

File: README.md [hunk 6]
```diff
@@ -65,9 +90,19 @@ What happens next is up to the language. There's three things that need to be re
 
 # How the request works
 
-The request that is made when `run` is called is a simple POST to the endpoint `https://api.readme.build/v1/run/{service}/{action}`. The body should be the JSON blob of data passed in, and the API key should be passed as basic auth, with the username being the API key and the password being blank. (So, `headers['Authorization'] = 'Basic ' + base64(key + ':');`, although most request libraries simplify this.)
+The request that is made when `run` is called is a simple POST to the endpoint `https://api.readme.build/v1/run/{service}/{action}`. The files being sent to the service should be sent as seperate form-data items. The rest of the json data should be stringified in the `data` form item. This guarantees that everything retains its correct type, sinceeverything in form-data is a string.
```
Comment (domharrington) [inline README.md]: Can you remove the word "simple"?

Comment (domharrington) [reply README.md]: sinceeverything needs a space

File: README.md [hunk 7]
```diff
@@ -105,3 +140,15 @@ Everywhere:
   * Will there be a hard deprecation header? or just an error?
   * How do we deal with non-JSON responses (like just an integer)
   * Do we handle team names yet? How so? Versions?
+
+# SDK Supported Features
+
+| SDK        | [Invoke](https://docs.readme.build/docs/getting-started-2) | Private Services | [Files](https://docs.readme.build/v1.0/docs/sending-files)   | Outputs |
```
Comment (domharrington) [inline README.md]: I hate tables in markdown.

File: README.md [hunk 8]
```diff
@@ -105,3 +140,15 @@ Everywhere:
...
+| ---------- | :----: | :--------------: | :-----: | :-----: |
+| [**Node**](https://github.com/readmeio/api) |✔|✔|✔|✔|
+| [**Python**](https://github.com/readmeio/api-python) |✔|✔|✕|✕|
+| [**Ruby**](https://github.com/readmeio/api-ruby) |✔|✔|✕|✕|
+| **Slack**  |✔|✔|✕|✕|
```
Comment (domharrington) [inline README.md]: Can you link to a docs page for slack and cron? Even if we dont have any content yet.

**Review by domharrington — COMMENTED** [review]
Looking good!

File: README.md (hunk 7 above)
Comment (mjcuva) [reply README.md]: Ugh same

File: README.md (hunk 5 above)
Comment (mjcuva) [reply README.md]: It's one I have locally, I can deploy it to prod once we merge all the file stuff

**Pull request merged**